neo4j/src/main/java/com/yahoo/ycsb/db/Neo4jClient.java

```diff
@@ -0,0 +1,214 @@
+/**
+ * Copyright (c) 2012 - 2016 YCSB contributors. All rights reserved.
+ * <p/>
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+package com.yahoo.ycsb.db;
+
+import com.yahoo.ycsb.ByteIterator;
+import com.yahoo.ycsb.DB;
+import com.yahoo.ycsb.DBException;
+import com.yahoo.ycsb.Status;
+import com.yahoo.ycsb.StringByteIterator;
+import org.neo4j.graphdb.*;
+import org.neo4j.graphdb.factory.GraphDatabaseFactory;
+import org.neo4j.helpers.collection.Iterators;
+
+import java.io.File;
+import java.util.*;
+
+/**
+ * Neo4j client for YCSB framework.
+ */
+public class Neo4jClient extends DB {
+  private Properties props;
+  public static final String DEFAULT_PROP = "";
+  public static final String BASE_PATH = "db.path";
+  private static final String NODE_ID = "_key";
+
+  private GraphDatabaseService graphDb;
+
+  /**
+   * Creates (if not existing yet) a graph database, and initializes it.
+   *
+   * @throws DBException
+   */
+  @Override
+  public void init() throws DBException {
+    // getting database path
+    props = getProperties();
+    String basePath = props.getProperty(BASE_PATH, DEFAULT_PROP);
+
+    graphDb = new GraphDatabaseFactory().newEmbeddedDatabase(new File(basePath));
```

Comment:
If we run multiple client threads, will this properly handle creating the embedded database only once?